extensions/wikia/FacebookClient/FacebookClientController.class.php

```diff
@@ -151,21 +151,18 @@ public function disconnectFromFB() {
 
 		FacebookMapModel::deleteFromWikiaID( $user->getId() );
 
-		$params = new FauxRequest( [ 'wpName' => $user->getName() ] );
-		$loginForm = new LoginForm( $params );
-
-		if ( $user->getOption( 'fbFromExist' ) ) {
-			$res = $loginForm->mailPasswordInternal( $user, true, 'fbconnect-passwordremindertitle-exist', 'fbconnect-passwordremindertext-exist' );
-		} else {
-			$res = $loginForm->mailPasswordInternal( $user, true, 'fbconnect-passwordremindertitle', 'fbconnect-passwordremindertext' );
-		}
+		// Create a temporary password for the user
+		$userService = new \UserService();
+		$tempPass = $userService->resetPassword( $user );
+
+		// Send email to user with temp password, telling them their FB account is disconnected
+		$emailParams = [
+			'targetUser' => $user,
+			'tempPass' => $tempPass,
+		];
+		F::app()->sendRequest( 'Email\Controller\FacebookDisconnect', 'handle', $emailParams );
```

Comment:
This does actually send a status back.  The ```sendRequest``` method returns a ```WikiaResponse``` object on which you can call ```getData()```.  This will be an array with a ```result``` key that will either be ```ok``` or some type of error key.  For errors, the error message will be in ```msg```.

We don't use this everywhere since a lot of places that send email didn't check the status of the email anyway (its a fire and forget process that is a side effect to something else the user was doing).  Here though, since this code actually does look for an error, we should probably keep and update it.